Jenkinsfile

```diff
@@ -57,111 +56,202 @@ spec:
     }
   }
   stages {
-    stage('Setting variables ') {
+    stage('Use last tag commit if present ') {
       when { buildingTag() }
       steps {
         script {
             IMAGE_TAG = "${TAG_NAME}"
+            sh (script: 'git tag', returnStdout: true).trim()
+            sh (script: 'git describe', returnStdout: true).trim()
         }
       }
     }
 
     stage('Run maven') {
-      when { branch 'master' }
       steps {
         container('maven') {
           sh 'mvn -Dmaven.test.failure.ignore clean package'
         }
       }
+      post {
+        success{
+            println "Maven build is OK"
+        }
+        failure{
+            println "Something wrong with maven build"
+        }
+      }
     }
 
     stage('Docker image build') {
-        parallel {
-            stage('PR docker build') {
-                when { changeRequest target: 'master' }
-                    steps {
-                        container('docker') {
+        environment {
+            GREETING="${IMAGE_TAG}"
+        }
+        steps {
+            container('docker') {
+                script {
+                    IMAGE_ID = sh (script: 'docker build . -q --build-arg GREETING', returnStdout: true).trim()
```

Comment:
have you tested this?
I think it should be double quotes `"$GREETING"` to interpolate variable

Reply:
Yes. It works.